sycl/test-e2e/ESIMD/unified_memory_api/Inputs/prefetch.hpp

```diff
@@ -378,3 +396,164 @@ template <typename T> bool testACC(queue Q) {
+                              LoadPropertiesT LoadProperties) {
+
+  uint32_t Size = Groups * Threads * N;
+  std::cout << "BlockLoad USM case: T=" << esimd_test::type_name<T>()
+            << ",N=" << N;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{Threads};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  constexpr size_t Alignment = getAlignment<T, N, true>(LoadProperties);
+  T *In = sycl::aligned_alloc_shared<T>(Alignment, Size, Q);
+  T *Out = sycl::aligned_alloc_shared<T>(Alignment, Size, Q);
+  for (int i = 0; i < Size; i++) {
+    In[i] = esimd_test::getRandomValue<T>();
+    Out[i] = 0;
+  }
+
+  try {
+    Q.submit([&](handler &CGH) {
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> NDI) SYCL_ESIMD_KERNEL {
+         uint16_t GlobalID = NDI.get_global_id(0);
+         uint32_t ElemOffset = GlobalID * N;
+
+         simd<T, N> Vals;
+         simd_mask<1> Mask = 1;
+
+         if (GlobalID & 0x1)
+           prefetch<T, N>(In + ElemOffset, Mask, LoadPropertiesT{});
+         else
+           prefetch<T, N>(In, ElemOffset * sizeof(T), Mask, LoadPropertiesT{});
+
+         Vals = block_load<T, N>(In + ElemOffset, Mask, LoadPropertiesT{});
+
+         Vals.copy_to(Out + ElemOffset);
+       });
+     }).wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    sycl::free(In, Q);
+    sycl::free(Out, Q);
+    return false;
+  }
+
+  bool Passed = verifyBlockLoad<T>(In, Out, Size, N);
+  sycl::free(In, Q);
+  sycl::free(Out, Q);
+  return Passed;
+}
+
+template <typename T> bool testBlockLoadPrefetchUSM(queue Q) {
+
+  bool Passed = true;
+
+  // Using mask or cache hints adds the requirement to run tests on DG2/PVC.
+  // Also, DG2/DG2/PVC variant currently requires a) power-or-two elements,
+  // b) the number of bytes loaded per call must not exceed 512,
+  // c) the alignment of USM ptr + offset to be 4 or 8-bytes(for 8-byte
+  // element vectors).
+
+  constexpr size_t RequiredAlignment = sizeof(T) <= 4 ? 4 : 8;
+  properties DG2OrPVCProps{cache_hint_L1<cache_hint::streaming>,
+                           cache_hint_L2<cache_hint::cached>,
+                           alignment<RequiredAlignment>};
+
+  // Only d/q-words are supported now.
+  // Thus we use this I32Factor for testing purposes and convenience.
+  constexpr int I32Factor =
+      std::max(static_cast<int>(sizeof(int) / sizeof(T)), 1);
+  Passed &= testBlockLoadPrefetchUSM<T, 1 * I32Factor>(Q, 2, 4, DG2OrPVCProps);
+  Passed &= testBlockLoadPrefetchUSM<T, 2 * I32Factor>(Q, 5, 5, DG2OrPVCProps);
+  Passed &= testBlockLoadPrefetchUSM<T, 4 * I32Factor>(Q, 5, 5, DG2OrPVCProps);
+  Passed &= testBlockLoadPrefetchUSM<T, 8 * I32Factor>(Q, 5, 5, DG2OrPVCProps);
+  Passed &= testBlockLoadPrefetchUSM<T, 16 * I32Factor>(Q, 5, 5, DG2OrPVCProps);
+  Passed &= testBlockLoadPrefetchUSM<T, 32 * I32Factor>(Q, 2, 4, DG2OrPVCProps);
```

Comment:
Please check that 256-elements for words and 256/512-elements for bytes can be prefetched.
That may require splitting of the prefetch_usm_dg2_pvc.cpp to 2 tests - 1 for DG2 and 1 for PVC.